core/src/main/java/jenkins/triggers/ReverseBuildTrigger.java

```diff
@@ -113,7 +114,12 @@ private boolean shouldTrigger(Run upstreamBuild, TaskListener listener) {
             return false;
         }
         // This checks Item.READ also on parent folders; note we are checking as the upstream auth currently:
-        boolean downstreamVisible = jenkins.getItemByFullName(job.getFullName()) == job;
+        boolean downstreamVisible = false;
+        try {
+            downstreamVisible = jenkins.getItemByFullName(job.getFullName()) == job;
+        } catch (AccessDeniedException ex) {
+            // ignore, we will fall-back later
```

Comment:
To be fully consistent we could actually distinguish both and log a different warning in each case:

{code}
...
        try {
            downstreamVisible = jenkins.getItemByFullName(job.getFullName()) == job;
        } catch (AccessDeniedException ex) {
            downstreamDiscoverable = false;
        }
...
{code}